src/dune/exe.ml

```diff
@@ -152,28 +153,49 @@ let link_exe ~loc ~name ~(linkage : Linkage.t) ~cm_files ~link_time_code_gen
      in
      let open Build.With_targets.O in
      Build.with_no_targets prefix
-     >>> Command.run ~dir:(Path.build ctx.build_dir)
-           (Context.compiler ctx mode)
-           [ Command.Args.dyn ocaml_flags
-           ; A "-o"
-           ; Target exe
-           ; As linkage.flags
-           ; Dyn link_args
-           ; Command.of_result_map link_time_code_gen
-               ~f:(fun { Link_time_code_gen.to_link; force_linkall } ->
-                 S
-                   [ As
-                       ( if force_linkall then
-                         [ "-linkall" ]
-                       else
-                         [] )
-                   ; Lib.Lib_and_module.L.link_flags to_link
-                       ~lib_config:ctx.lib_config ~mode:linkage.mode
-                   ])
-           ; Deps o_files
-           ; Dyn (Build.map top_sorted_cms ~f:(fun x -> Command.Args.Deps x))
-           ; Fdo.Linker_script.flags fdo_linker_script
-           ])
+     >>> let+ cmd_run =
+           Command.run ~dir:(Path.build ctx.build_dir)
+             (Context.compiler ctx mode)
+             [ Command.Args.dyn ocaml_flags
+             ; A "-o"
+             ; Target exe
+             ; As linkage.flags
+             ; Dyn link_args
+             ; Command.of_result_map link_time_code_gen
+                 ~f:(fun { Link_time_code_gen.to_link; force_linkall } ->
+                   S
+                     [ As
+                         ( if force_linkall then
+                           [ "-linkall" ]
+                         else
+                           [] )
+                     ; Lib.Lib_and_module.L.link_flags to_link
+                         ~lib_config:ctx.lib_config ~mode:linkage.mode
+                     ])
+             ; Deps o_files
+             ; Dyn (Build.map top_sorted_cms ~f:(fun x -> Command.Args.Deps x))
+             ; Fdo.Linker_script.flags fdo_linker_script
+             ]
+         and+ cbi =
+           match custom_build_info with
+           | None -> Build.With_targets.return Action.empty
+           | Some { Custom_build_info.action = loc, action; _ } ->
+             let filename =
```

Comment:
is the trip through a temporary file necessary?

Reply:
When substitution happens at install-time there is not a lot of context left:

https://github.com/ocaml/dune/blob/master/bin/install_uninstall.ml#L195-L216